src/phases/pokemon-heal-phase.ts

```diff
@@ -121,8 +121,8 @@ export class PokemonHealPhase extends CommonAnimPhase {
       }
 
       pokemon.updateInfo().then(() => super.end());
-    } else if (this.healStatus && !this.revive && pokemon.status) {
-      lastStatusEffect = pokemon.status.effect;
+      lastStatusEffect = pokemon.getStatusEffect(true);
+    } else if (this.healStatus && !this.revive && lastStatusEffect !== StatusEffect.NONE) {
```

Comment:
I don't think this is correct, you're setting the lastStatusEffect in another else if, but if you're in that else if you're never getting to the one using the value
```suggestion
        } else if (this.healStatus && !this.revive && pokemon.hasStatusEffect(false, true)) {
      lastStatusEffect = pokemon.getStatusEffect(true);
```